### PR TITLE
Add support for using GPS heading on multirotors

### PR DESCRIPTION
**Pull request by fiam**
Uses the same logic as FW GPS heading, but applies a correction to
the heading reported by the GPS by using the tilt angle of the MR
head.

Please, be very careful testing this feature for the moment.

This commit introduces the new setting use_gps_heading, with
the following values:

- AUTO (default): Enabled just for FW
- ON: Enabled for all crafts
- OFF: Disabled for all crafts

### CODE REVIEW

File: src/main/flight/imu.c [hunk 1]
```diff
@@ -457,41 +457,78 @@ static void imuCalculateEstimatedAttitude(float dT)
...
+                groundCourse = gpsSol.groundCourse;
+            } else {
+                // MR might fly on any direction, but it tends to fly in the direction the
+                // head it's tilted to. To compensate for this, we rotate gpsSol.groundCourse
+                // by the tilt direction. Pitch and roll angles must fall in the [10, 90) so
+                // the data doesn't get polluted during flips or rolls.
+                // For now, we also require a harcoded speed of 3m/s, but we
+                // should adjust this depending on the maximum speed in modes which do limit
+                // it (e.g. RTH).
+                const uint16_t minAttitudeAngle = DECIDEGREES_TO_DEGREES(10);
```
Comment (digitalentity) [inline src/main/flight/imu.c]: You can check `calculateCosTiltAngle()` which is essentially generalized tilt `cos(roll)*cos(pitch)`

File: src/main/flight/imu.c [hunk 2]
```diff
@@ -457,41 +457,78 @@ static void imuCalculateEstimatedAttitude(float dT)
...
-        bool canUseCOG = sensors(SENSOR_GPS) && STATE(GPS_FIX) && gpsSol.numSat >= 6 && gpsSol.groundSpeed >= 300;
+    bool canUseCOG = false;
+    uint16_t groundCourse;
+    if (sensors(SENSOR_GPS) && STATE(GPS_FIX) && gpsSol.numSat >= 6) {
+        // TODO: Use GPS heading if the real compass fails during a flight
+        switch (imuConfig()->use_gps_heading) {
+            case GPS_HEADING_AUTO:
+                canUseCOG = STATE(FIXED_WING);
+                break;
+            case GPS_HEADING_ENABLED:
```
Comment (digitalentity) [inline src/main/flight/imu.c]: We need to verify that MAG is preferred if available.

**Review by digitalentity — COMMENTED** [review]
Nice!

**Comment by fiam** [comment]
Would fix #2688, could be also expanded to fix  #1323

**Pull request closed without merging**